sdk/storage/Azure.Storage.Blobs.Cryptography/src/ClientSideDecryptionPolicy.cs

```diff
@@ -0,0 +1,412 @@
+            var encryptionData = GetAndValidateEncryptionData(metadata);
+
+            Stream plaintext;
+            int read = 0;
+            if (encryptionData.HasValue)
+            {
+                byte[] IV;
+                if (encryptedBlobRange.AdjustedRange.Offset == 0)
+                {
+                    IV = encryptionData.Value.ContentEncryptionIV;
+                }
+                else
+                {
+                    IV = new byte[EncryptionConstants.ENCRYPTION_BLOCK_SIZE];
+                    await ciphertext.ReadAsync(IV, 0, IV.Length).ConfigureAwait(false);
+                    read = IV.Length;
+                }
+
+                plaintext = WrapStream(
+                    ciphertext,
+                    (await GetContentEncryptionKeyAsync(encryptionData.Value).ConfigureAwait(false)).ToArray(),
+                    encryptionData.Value,
+                    IV,
+                    noPadding);
+            }
+            else
+            {
+                plaintext = ciphertext;
+            }
+
+            // still need to readjust ranges even if we didn't decrypt anything, so keep this out of branch
+            int gap;
+            if ((gap = (int)(encryptedBlobRange.OriginalRange.Offset - encryptedBlobRange.AdjustedRange.Offset) - read) > 0)
+            {
+                await plaintext.ReadAsync(new byte[gap], 0, gap).ConfigureAwait(false);
+            }
+
+            return new LengthLimitingStream(plaintext, encryptedBlobRange.OriginalRange.Count);
+        }
+
+        #endregion
+
+        private static Metadata ExtractMetadata(ResponseHeaders headers)
+        {
+            Metadata metadata = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (HttpHeader header in headers)
+            {
+                if (header.Name.StartsWith(Constants.HeaderNames.MetadataPrefix, StringComparison.InvariantCulture))
+                {
+                    metadata[header.Name.Substring(Constants.HeaderNames.MetadataPrefix.Length)] = header.Value;
+                }
+            }
+
+            return metadata;
+        }
+
+        /// <summary>
+        /// Gets and validates a blob's encryption-related metadata
+        /// </summary>
+        /// <param name="metadata">The blob's metadata</param>
+        /// <returns>The relevant metadata.</returns>
+        private static EncryptionData? GetAndValidateEncryptionData(Metadata metadata)
+        {
+            _ = metadata ?? throw new InvalidOperationException();
+
+            EncryptionData encryptionData;
+            if (!metadata.TryGetValue(EncryptionConstants.ENCRYPTION_DATA_KEY, out string encryptedDataString))
+            {
+                return default;
+            }
+
+            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(encryptedDataString)))
+            {
+                var serializer = new DataContractJsonSerializer(typeof(EncryptionData));
```

Comment:
We don't use `DataContractJsonSerializer` any more -- we use the `System.Text.Json` library now.  I don't think you need to fix this right now, but please file a bug on removing this.